target/linux/ramips/image/mt7621.mk

```diff
@@ -250,6 +250,18 @@ define Device/dlink_dir-8xx-a1
 	check-size
 endef
 
+define Device/dlink_dir-8xx-r1
+  $(Device/dsa-migration)
+  IMAGE_SIZE := 16064k
+  DEVICE_VENDOR := D-Link
+  DEVICE_PACKAGES := kmod-mt7615e kmod-mt7615-firmware
+  KERNEL_INITRAMFS := $$(KERNEL) 
+  IMAGES += factory.bin
+  IMAGE/sysupgrade.bin := append-kernel | append-rootfs |\
+	pad-rootfs | append-metadata | check-size
+  IMAGE/factory.bin := append-kernel | append-rootfs | check-size
+endef
```

Comment:
Do you expect other similar devices? If not, just put the command into the device's node as long as only one device inherits from it.

Reply:
I think that the DIR-878-R1 will have the same partition structure and can reuse these files.